once/__init__.py

```diff
@@ -182,26 +233,24 @@ def _inspect_function(self, func: collections.abc.Callable):
     # bound version of the function to the object or class.
     def __get__(self, obj, cls):
         if self.is_classmethod:
-            func = functools.partial(self.func, cls)
-            return functools.partial(self._execute_call_once, func)
+            return self._callable(functools.partial(self.func, cls))
         if self.is_staticmethod:
-            return functools.partial(self._execute_call_once, self.func)
-        return functools.partial(self._execute_call_once, self.func, obj)
+            return self._callable(self.func)
+        return self._callable(functools.partial(self.func, obj))
 
 
 class once_per_instance(_OnceBase):  # pylint: disable=invalid-name
     """A version of once for class methods which runs once per instance."""
 
     def __init__(self, func: collections.abc.Callable) -> None:
-        super().__init__(func)
-        self.return_value: weakref.WeakKeyDictionary[
-            typing.Any, typing.Any
+        self.func = self._inspect_function(func)
+        super().__init__(_wrapped_function_type(self.func))
+        self.callables_lock = threading.Lock()
+        self.callables: weakref.WeakKeyDictionary[
+            typing.Any, collections.abc.Callable
```

Comment:
If using typing.Any, would it be a good idea to add a comment what we expect here? Iiuc, its an object of type Class Instance this decorator is on right?

Reply:
done - added a comment